### PR TITLE
Fixed 8 coverity bugs

### PR DESCRIPTION
**Pull request by wargio**
* 1372258 Logically dead code
* 1372262 Logically dead code
* 1372264 Logically dead code
* 1372265 Dereference after null check
* 1372272 Logically dead code
* 1372273 Logically dead code
* 1372276 Dereference after null check
* 1372278 Identical code for different branches

### CODE REVIEW

**Comment by XVilka** [comment]
I suggest we wait until master is green. Fixing coverities sometimes can lead to segfaults.

**Comment by wargio** [comment]
i can assure that these cannot lead to segfaults.
currently, without the patch, these bugs lead to segfaults.

File: libr/asm/asm.c [hunk 1]
```diff
@@ -307,8 +307,8 @@ static int has_bits(RAsmPlugin *h, int bits) {
 R_API void r_asm_set_cpu(RAsm *a, const char *cpu) {
 	if (a) {
 		free (a->cpu);
+		a->cpu = cpu? strdup (cpu): NULL;
```
Comment (radare) [inline libr/asm/asm.c]: r_str_dup checks for null, so you can skip a ternary op here

Comment (wargio) [reply libr/asm/asm.c]: I'll make a patch.

**Pull request merged**